tests/unit/test_endpoint.py

```diff
@@ -239,16 +242,16 @@ def test_retry_events_can_alter_behavior(self):
             expected_events=[
                 'request-created.ec2.DescribeInstances',
                 'before-send.ec2.DescribeInstances',
+                'before-parse.ec2.DescribeInstances',
                 'response-received.ec2.DescribeInstances',
                 'needs-retry.ec2.DescribeInstances',
             ]
             * 2,
         )
 
     def test_retry_on_socket_errors(self):
-        self.event_emitter.emit.side_effect = self.get_emitter_responses(
-            num_retries=1
-        )
+        val = self.get_emitter_responses(num_retries=1, num_events=3)
```

Comment:
This is a leftover from some testing; we should revert this change, except for the num_events part